torch/csrc/jit/docs/serialization.md

```diff
@@ -0,0 +1,237 @@
+# Torchscript serialization
+
+This document explains the Torchscript serialization format, and the anatomy of a call to `torch::jit::save()` or `torch::jit::load()`.
+
+## Overview
+
+A serialized model (call it `model.pt`) is a ZIP archive containing many files. It can be directly inspected by calling `unzip` on it. The archive's file structure looks like:
+
+```
+model.pt
+|-- model.json
+|-- code/
+    |-- __torch__.py
+    |-- __torch__.py.debug_pkl
+    |-- foo/
+        |-- bar.py
+        |-- bar.py.debug_pkl
+|-- data.pkl
+|-- tensors/
+    |-- 0
+    |-- 1
+```
+
+You'll notice that there are `.py` and `.pkl` files in this archive. That's because our serialization format tries to mimic Python's. All "code-like" information (methods, modules, classes, functions) are stored as human-readable `.py` containing valid Python syntax, and all "data-like" information (attributes, objects, etc.) are pickled using a subset of Python's pickle protocol.
+
+A model is really a top-level module with some submodules, parameters, and so on depending on what the author needs. So, `data.pkl` contains the pickled top-level module. Deserializing the model is as simple as calling `unpickle()` on `data.pkl`, which will restore the module object with its associated code and data.
```

Comment:
> Deserializing the model is as simple as calling `unpickle()` on `data.pkl`

What confuses me in this paragraph is that you were previously saying that pickle is only for data - how do we pick up the code as well from this? Some clarification might be helpful.

Reply:
Thanks for the feedback! I will add a section on how models are loaded, hopefully that will clarify these questions